### PR TITLE
feat: add desciptions and covers

### PR DESCRIPTION
**Pull request by GeekaN2**
Look at this DUDE

image

image

image

### CODE REVIEW

**Comment by slaveeks** [comment]
conflict

File: src/presentation/components/marketplace/EditorToolElement.vue [hunk 1]
```diff
@@ -1,10 +1,9 @@
 <template>
   <Card
     :title="tool.title"
-    subtitle="
-Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam maximus odio non lectus maximus suscipit. Morbi vehicula neque eget augue mollis porta. Sed cursus auctor nisi a iaculis."
+    :subtitle="tool.description"
     orientation="horizontal"
-    src="https://sun9-50.userapi.com/c844720/v844720274/194ada/1HCPufLxhzY.jpg"
+    :src="`http://localhost:9001/api/v1/buckets/editor-tool-covers/objects/download?preview=true&prefix=${tool.cover}&version_id=null`"
```
Comment (slaveeks) [inline src/presentation/components/marketplace/EditorToolElement.vue:6]: not production thing

File: src/infrastructure/transport/notes-api/index.ts [hunk 2]
```diff
@@ -86,10 +86,10 @@ export default class NotesApiTransport extends AuthorizableTransport {
    * @param data - data to be sent with request body
    * @param params - Additional params to tune request
    */
-  public async post<Payload>(endpoint: string, data?: JSONValue | FormData, params?: NotexApiRequestParams): Promise<Payload> {
+  public async post<Payload, Response = Payload>(endpoint: string, data?: JSONValue | FormData, params?: NotexApiRequestParams): Promise<Response> {
```
Comment (slaveeks) [inline src/infrastructure/transport/notes-api/index.ts:89]: image

File: src/presentation/components/marketplace/EditorToolElement.vue (hunk 1 above)
Comment (GeekaN2) [reply src/presentation/components/marketplace/EditorToolElement.vue:6]: Waiting for Nikita

**Review by slaveeks — APPROVED** [review]

**Review by neSpecc — CHANGES_REQUESTED** [review]
Home page is broken 

image

File: codex-ui/src/styles/dimensions.pcss [hunk 3]
```diff
@@ -23,6 +23,7 @@
    * Layout
    */
   --layout-content-width: 700px;
+  --layout-block-width: 268px;
```
Comment (neSpecc) [inline codex-ui/src/styles/dimensions.pcss:26]: left is 300. there are also 16 px paddings 
image

File: src/presentation/pages/marketplace/MarketplacePage.vue [hunk 4]
```diff
@@ -0,0 +1,57 @@
+<template>
+  <DefaultLayout>
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/MarketplacePage.vue:2]: Lets call it `ThreeColsLayout`

File: src/presentation/pages/marketplace/MarketplacePage.vue [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    <template #block-left>
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/MarketplacePage.vue:3]:
```suggestion
    <template #left-col>
```

File: src/presentation/pages/marketplace/MarketplacePage.vue [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      <VerticalMenu
+
+        class="menu"
+        :items="verticalMenuItems"
+      />
+    </template>
+    <template #default>
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/MarketplacePage.vue:10]:
```suggestion
    <template #center>
```

File: src/presentation/pages/marketplace/MarketplacePage.vue [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      <Marketplace v-if="route.path === '/marketplace'" />
+      <AddTool v-if="route.path === '/marketplace/add'" />
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/MarketplacePage.vue:12]: use sub routes instead

File: src/presentation/pages/marketplace/AddTool.vue [hunk 8]
```diff
@@ -71,24 +135,78 @@ function onClick() {
...
+    cover: cover.value,
     source: {
       cdn: toolCdn.value,
     },
   }).then(() => {
     router.push({ path: '/marketplace' });
   });
 }
+
+async function setCover(event: Event) {
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/AddTool.vue:149]: docs missed

**Review by neSpecc — COMMENTED** [review]
Paddings are different

image

File: src/infrastructure/marketplace.repository.ts [hunk 9]
```diff
@@ -37,7 +37,25 @@ export default class MarketplaceRepository implements MarketplaceRepositoryInter
    * @param tool - tool data
    */
   public async addTool(tool: NewToolData): Promise<EditorTool> {
-    const res = await this.transport.post<{ data: EditorTool }>('/editor-tools/add-tool', tool);
+    const data = new FormData();
```
Comment (neSpecc) [inline src/infrastructure/marketplace.repository.ts:40]: it is a transport-level staff, repo should not care about it. Move it to the transport

File: src/presentation/pages/marketplace/AddTool.vue [hunk 10]
```diff
@@ -1,58 +1,117 @@
 <template>
   <div class="container">
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/AddTool.vue:2]: use large `dimensions`

File: src/presentation/pages/marketplace/AddTool.vue [hunk 11]
```diff
@@ -1,58 +1,117 @@
...
-      :title="$t('marketplace.newTool.exportName.label')"
-      :caption="$t('marketplace.newTool.exportName.caption')"
-    >
-      <Input
-        v-model:model-value="toolExport"
-        name="toolExport"
-        :placeholder="$t('marketplace.newTool.exportName.placeholder')"
-      />
-    </Section>
+    <span class="headings">
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/AddTool.vue:3]: why span?

File: src/presentation/pages/marketplace/AddTool.vue [hunk 12]
```diff
@@ -1,58 +1,117 @@
...
+    </span>
+    <div class="section-group">
+      <Heading
+        :level="3"
+        class="headings"
+      >
+        {{ $t("marketplace.userPerspective") }}
+      </Heading>
+      <div class="section">
+        <Section
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/AddTool.vue:17]: We have Fieldset component for the `title+sections`

File: src/presentation/pages/marketplace/AddTool.vue [hunk 13]
```diff
@@ -1,58 +1,117 @@
...
+        >
+          <Input
+            v-model:model-value="description"
+            name="description"
+            :placeholder="$t('marketplace.newTool.description.placeholder')"
+          />
+        </Section>
+        <Section :title="$t('marketplace.newTool.picture.label')">
+          <Card
+            subtitle="A cool descriptive screenshot of the Tool's UI"
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/AddTool.vue:39]: i18n missed

File: src/presentation/pages/marketplace/AddTool.vue [hunk 14]
```diff
@@ -1,58 +1,117 @@
...
+              hidden
+              accept="image/*"
+              enctype="multipart/form-data"
+              @change="setCover($event)"
+            >
+            <label
+              for="marketplace-new-tool-cover"
+              class="add-cover-label"
+            >
+              <Button class="add-cover-button"> Select file </Button>
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/AddTool.vue:56]: i18n

File: src/presentation/pages/marketplace/AddTool.vue [hunk 15]
```diff
@@ -71,24 +135,78 @@ function onClick() {
...
     },
   }).then(() => {
     router.push({ path: '/marketplace' });
   });
 }
+
+async function setCover(event: Event) {
+  cover.value = (event.target as HTMLInputElement).files?.[0];
+
+  function arrayBufferToBase64(buffer: ArrayBuffer) {
```
Comment (neSpecc) [inline src/presentation/pages/marketplace/AddTool.vue:152]: move to `infrastructure/utils/buffer`

File: src/domain/noteList.service.ts [hunk 16]
```diff
@@ -50,6 +50,7 @@ export default class NoteListService {
       let objUrl: string | null = null;
 
       try {
+        console.log('bonka', note?.id, note.cover);
```
Comment (neSpecc) [inline src/domain/noteList.service.ts:53]:
```suggestion
```

File: src/infrastructure/transport/types/FileDto.ts [hunk 17]
```diff
@@ -0,0 +1,10 @@
+type FileDto = {
+  key: string;
+  blob: Blob;
+  fileName: string;
```
Comment (neSpecc) [inline src/infrastructure/transport/types/FileDto.ts:4]: add docs

File: src/infrastructure/transport/notes-api/index.ts [hunk 18]
```diff
@@ -85,11 +86,38 @@ export default class NotesApiTransport extends AuthorizableTransport {
...
    * @param params - Additional params to tune request
+   * @param files - Files to be sent with request
    */
-  public async post<Payload>(endpoint: string, data?: JSONValue | FormData, params?: NotexApiRequestParams): Promise<Payload> {
-    const response = await super.post(endpoint, data, params);
+  public async post<Response>(endpoint: string, data?: Record<string, JSONValue | undefined>, params?: NotexApiRequestParams, files?: FilesDto): Promise<Response> {
+    let payload: FormData | Record<string, JSONValue | undefined> = data ?? {};
 
-    return response as Payload;
+    if (files) {
```
Comment (neSpecc) [inline src/infrastructure/transport/notes-api/index.ts:94]: move to fetch transport

File: src/infrastructure/transport/notes-api/index.ts [hunk 19]
```diff
@@ -85,11 +86,38 @@ export default class NotesApiTransport extends AuthorizableTransport {
...
+      const multipartFormData = new FormData();
```
Comment (neSpecc) [inline src/infrastructure/transport/notes-api/index.ts:95]: create util `composeFormDataFromFilesDto()`

File: src/application/router/routes.ts [hunk 20]
```diff
@@ -89,20 +90,31 @@ const routes: RouteRecordRaw[] = [
...
     },
-  },
-  {
-    name: 'newTool',
-    path: `/marketplace/add`,
-    component: AddTool,
-    meta: {
-      pageTitleI18n: 'pages.addTool',
+    children: [{
+      name: 'marketplace',
```
Comment (neSpecc) [inline src/application/router/routes.ts:101]:
```suggestion
      name: 'marketplaceTools',
```

File: src/application/router/routes.ts [hunk 21]
```diff
@@ -89,20 +90,31 @@ const routes: RouteRecordRaw[] = [
...
+      path: 'tools',
+      component: Marketplace,
+      meta: {
+        pageTitleI18n: 'pages.marketplace',
+      },
     },
+    {
+      name: 'newTool',
```
Comment (neSpecc) [inline src/application/router/routes.ts:109]:
```suggestion
      name: 'marketplaceAddTool',
```

File: src/infrastructure/transport/fetch.transport.ts [hunk 22]
```diff
@@ -74,11 +76,23 @@ export default class FetchTransport {
...
-    const isFormData = payload !== undefined && payload instanceof FormData;
+  public async post({
+    endpoint,
+    payload,
+    files,
+  }: {
+    endpoint: string;
+    payload?: Record<string, JSONValue | undefined>;
+    files?: FilesDto;
+  }): Promise<JSONValue> {
```
Comment (neSpecc) [inline src/infrastructure/transport/fetch.transport.ts:89]: move this object to types

File: src/infrastructure/transport/fetch.transport.ts [hunk 23]
```diff
@@ -74,11 +76,23 @@ export default class FetchTransport {
...
+    const isFormData = Boolean(files);
+    let body: FormData | string | undefined = payload ? JSON.stringify(payload) : undefined;
+
+    if (files) {
+      body = composeFormDatFromFilesDto(payload, files);
```
Comment (neSpecc) [inline src/infrastructure/transport/fetch.transport.ts:94]:
```suggestion
      body = composeFormDataFromFilesDto(payload, files);
```

File: src/infrastructure/transport/utils/composeFormDataFromFilesDto.ts [hunk 24]
```diff
@@ -0,0 +1,26 @@
+import type { FilesDto } from '../types/FileDto';
+import type JSONValue from '../types/JSONValue';
+
+export const composeFormDatFromFilesDto = (payload: Record<string, JSONValue | undefined> | undefined, files: FilesDto): FormData => {
```
Comment (neSpecc) [inline src/infrastructure/transport/utils/composeFormDataFromFilesDto.ts:4]: add cool docs please

File: src/infrastructure/utils/buffer.ts [hunk 25]
```diff
@@ -0,0 +1,11 @@
+export const arrayBufferToBase64 = (buffer: ArrayBuffer): string => {
```
Comment (neSpecc) [inline src/infrastructure/utils/buffer.ts:1]: add simple docs

**Pull request merged**